hms-api/src/service/idea_comment-service.ts

```diff
@@ -0,0 +1,130 @@
+import Uuid from '../util/Uuid';
+import NotFoundError from '../error/NotFoundError';
+import {getIdea, ideaExists} from '../repository/idea-repository';
+import {getUser, userExists} from '../repository/user-repository';
+import {
+  commentAlreadyExists,
+  getComment,
+  putComment,
+  deleteComment,
+  listComments,
+} from '../repository/idea_comment-repository';
+import ValidationError from '../error/ValidationError';
+import Comment from '../repository/domain/IdeaComment';
+import ReferenceNotFoundError from '../error/ReferenceNotFoundError';
+import CommentResponse from '../rest/Comment/CommentResponse';
+import CommentDeleteResponse from '../rest/Comment/CommentDeleteResponse';
+import CommentListResponse from '../rest/Comment/CommentListResponse';
+import ideaComment from '../repository/domain/IdeaComment';
+
+export async function createComment(
+  ideaId: Uuid,
+  userId: Uuid,
+  text: string,
+  replyTo: Uuid,
+): Promise<Comment> {
+  if (!(await ideaExists(ideaId))) {
+    throw new ReferenceNotFoundError(`Idea with id: ${ideaId} not found`);
+  } else if (
+    replyTo &&
+    !(await commentAlreadyExists(new Comment(userId, ideaId, text, replyTo)))
+  ) {
+    throw new ReferenceNotFoundError(`Comment with id: ${replyTo} not found`);
+  } else if (!(await userExists(userId))) {
+    throw new ReferenceNotFoundError(`User with id: ${userId} not found`);
+  }
+
+  const comment = new Comment(userId, ideaId, text, replyTo);
+  const result = comment.validate();
+  if (result.hasFailed()) {
+    throw new ValidationError('Cannot create Comment', result);
+  }
+
+  await putComment(comment);
+
+  return comment;
```

Comment:
so a few things about this function:
1. You don't need them as `else if` statements because you throw inside each of them so you don't need to worry about them executing when they shouldn't (I get it makes it a little bit "shorter" like this but I would argue it is more readable as just a bunch of `if` statements)
2. `commentAlreadyExists(new Comment...` will never evaluate to true because when you make a new Comment it gets a new random ID (which for sure won't be in DynamoDB) so I would say this check isn't necessary here.  It WOULD be necessary if we had some other way of determining that two comments are the "same" (like identical text or something).  But as that isn't the case we could drop the check here.
3. You create a `new Comment` multiple times in this function, and they will each have different IDs (making them "different") as well it wastes a bit of memory and cpu cycles in doing so.  It would be better to just do it once, right before the first time you use it.
4. `Comment` as a class doesn't seem to exist anywhere, `IdeaComment` does however (I think you meant to use that)